Makefile

```diff
@@ -21,32 +21,36 @@ deps_table_update:
 
 # Check that source code meets quality standards
 
-extra_quality_checks: deps_table_update
+extra_quality_checks:
 	python utils/check_copies.py
 	python utils/check_table.py
 	python utils/check_dummies.py
 	python utils/check_repo.py
-	python utils/style_doc.py src/transformers docs/source --max_len 119
 	python utils/class_mapping_update.py
```

Comment:
should we create a new target for auto-generation? It'd be:
```
autogenerate_code: deps_table_update
        python utils/class_mapping_update.py
```

Reply:
Oh I hadn't realized `class_mapping_update` was writing things. It should also disappear from `make quality` then to go with the others in `extra_style_checks`.

For me:

`make fixup` -> applies the styling patches and checks everything passes quickly
`make style`  -> applying styling patches and other auto-generated code
`make quality` -> only checks, does not write anything

Reply:
I only ever use `make fixup` - the initial design was that `fixup` = `style+quality` on only modified files.

So it was created before we added 2 items that generated code, so if others prefer the quick version then `fixup` should include the autogenerate target too.

Actually, is there is a situation where `fixup` won't do the right thing already? When does one need to run the full `make style`?

Reply:
class_mapping_update doesn't do anything if the relevant file hasn't changed so it's blazingly fast
we can do the same for deps_table_update so it'd be instant 99% of the time.

Reply:
Yes `fixup` is `style+quality` on only modified files. My problem is that some processes that modify files have been added to `quality-extra-checks` which is run by `quality` on top of `fixup`. So `fixup` runs `extra-quality-checks` as well as `extra-styling-checks` to do both, I don't want to change that part.

Reply:
See latest version for my proposal:
- `extra-quality-checks` common processes for quality and fixup, checks that are fast or need to run on all files, not changing any file
- `extra-style-checks` common processes for style and fixup, checks that are fast or need to run on all files, changing files if necessary
- `style` style all files + other checks that are "writing" the right thing
- `quality` check all files + other checks (in check only mode)
-  `fixup` same as `style` + `quality` but only on modified files.